test/integration/targets/postgresql/tasks/main.yml

```diff
@@ -724,6 +724,49 @@
       - "result_database.stdout_lines[-1] == '(1 row)'"
       - "'{{ db_user2 }}=Cc' in result_database.stdout"
 
+- name: Create function for test
+  become: True
+  become_user: "{{ pg_user }}"
+  postgresql_query:
+    login_user: "{{ db_user1 }}"
+    login_password: password
+    db: "{{ db_name }}"
+    query: CREATE FUNCTION public.a() RETURNS integer LANGUAGE SQL AS 'SELECT 2';
+
+- name: Grant execute to all functions
+  postgresql_privs:
+    type: "function"
+    state: "present"
+    privs: "EXECUTE"
+    roles: "{{ db_user2 }}"
+    objs: "ALL_IN_SCHEMA"
+    schema: "public"
+    db: "{{ db_name }}"
+    login: "{{ db_user1 }}"
+    password: "password"
+    host: "localhost"
+
+- name: Check that all functions have no execute privileges
+  become: True
+  become_user: "{{ pg_user }}"
+  shell: psql {{ db_name }} -c "SELECT proacl FROM pg_proc WHERE proname = 'a'" -t
+  register: result_database
+- assert:
```

Comment:
4. Add gaps before -assert, it's easier to examine the tests

Reply:
This actually was done on purpose, since they are closely related: we run a command and then check it.

Reply:
IMO, it’s easier to sort it out and, at least, a common way to separate highlevel bloks of code in playbooks.

Reply:
IMO, it’s easier to sort it out and, at least, a common way to separate highlevel bloks of code in playbooks.